### PR TITLE
jobid: use semi human-readable container IDs

### PR DESCRIPTION
**Pull request by xoebus**
Even though we don't expose these as part of our public interface it can be helpful for operators using container metrics collectors if our container IDs are somewhat human readable.

Name: `test-server.alt-test-server`
Old: `ORSXG5BNONSXE5TFOIXGC3DUFV2GK43UFVZWK4TWMVZA===`
New: `bpm-test-server.2ealt-test-server`

### CODE REVIEW

**Comment by xoebus** [comment]
@ansd What do you think?

File: src/bpm/jobid/encoding.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	alphanumeric = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+	symbols      = "_-" // . is not included because we use it for escaping
+	combined     = alphanumeric + symbols
+)
+
+// Encode encodes an arbitrary container name into a format which is a valid
+// container name for runc. The encoding must not be depended upon and can
+// change unexpectedly between versions of BPM.
+func Encode(name string) string {
+	id := []byte{}
```
Comment (xoebus) [inline src/bpm/jobid/encoding.go]: This can probably be just a string?

File: src/bpm/jobid/encoding.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+// original string. The encoding must not be depended upon and can change
+// unexpectedly between versions of BPM.
+func Decode(id string) (string, error) {
+	if strings.HasPrefix(id, "bpm-") {
+		id = id[4:]
+	} else {
+		return "", fmt.Errorf("invalid job ID (missing prefix): %q", id)
+	}
+
+	name := []byte{}
```
Comment (xoebus) [inline src/bpm/jobid/encoding.go]: This can also probably be just a string.

File: src/bpm/jobid/encoding.go (hunk 1 above)
Comment (christarazi) [reply src/bpm/jobid/encoding.go]: Yeah makes sense. The append operation is casting what's already a string to a byte slice, so it makes more sense to just operate on strings.

**Comment by xoebus** [comment]
Should link to runc source which gives container ID restrictions in a comment

**Comment by ansd** [comment]
@xoebus many thanks. Looks great! I will test the integration in Sysdig probably next week. But it should work with the `bpm-` prefix and with the human readable IDs.

**Pull request merged**